### PR TITLE
audio: copier: move host and dai to copier folder

### PR DESCRIPTION
**Pull request by btian1**
Host and dai are already been integrated into copier, it is not standalone modules, so move it to copier folder.

### CODE REVIEW

**Review by lgirdwood — COMMENTED** [review]
LGTM

**Comment by ranj063** [comment]
> Host and dai are already been integrated into copier, it is not standalone modules, so move it to copier folder.

But the host and DAI components for IPC3 are not the same as the copier

**Comment by btian1** [comment]
> components for IPC3 are not

How about further separate with ipc3/ipc4 folder for host/dai-legacy and host/dai-zephyr?

**Review by RanderWang — APPROVED** [review]
LGTM

File: src/ipc/ipc4/dai.c [hunk 1]
```diff
@@ -28,6 +27,7 @@
 #include <stdbool.h>
 #include <stddef.h>
 #include <stdint.h>
+#include <../audio/copier/dai_copier.h>
```
Comment (kv2019i) [inline src/ipc/ipc4/dai.c:30]: Please use "../audio/copier" so this is relative to this source file.

**Review by kv2019i — CHANGES_REQUESTED** [review]
Hmm, this looks a bit messy as we have non-IPC elements that don't fit very nicely under "copier/" folder. @lgirdwood Do we really need to move copier? I don't think any module developer will use this as an example for anything.

File: src/ipc/ipc4/dai.c (hunk 1 above)
Comment (btian1) [reply src/ipc/ipc4/dai.c:30]: I am still using: target_include_directories(sof_options INTERFACE ${RIMAGE_TOP}/src/include)
othewise, it should be:  "../../audio/copier"

these include path come from:
target_include_directories(SOF INTERFACE ${RIMAGE_TOP}/src/include)
target_include_directories(SOF INTERFACE ${SOF_SRC_PATH}/include)
in zephyr/CMakeList.txt.

**Comment by lgirdwood** [comment]
> Hmm, this looks a bit messy as we have non-IPC elements that don't fit very nicely under "copier/" folder. @lgirdwood Do we really need to move copier? I don't think any module developer will use this as an example for anything.

Can you give an example of non-IPC elements that are causing concern ?

**Comment by kv2019i** [comment]
@lgirdwood wrote:
> @kv2019i wrote:
> > Hmm, this looks a bit messy as we have non-IPC elements that don't fit very nicely under "copier/" folder. @lgirdwood Do we really need to move copier? I don't think any module developer will use this as an example for anything.
> 
> Can you give an example of non-IPC elements that are causing concern ?

Sorry, typo there, i mean "non-IPC3 elements". "dai-legacy.c" is not related to "copier" so "audio/copier/ipc3/dai-legacy.c" path is confusing. Also, "dai-zephyr.c" is not specific to IPC4, so having that in ipc4 folder is confusing (implying IPC4 and Zephyr would be coupled, which is not the case).

**Comment by btian1** [comment]
> ements that a
@kv2019i your concern is reasonable, host and dai are not always exactly included by copier.

I also realized this, it is not easy to centralize these code, below is some possible option:
1. create two new folder in src/audio with name host / dai, and move all related code to src/audio/host src/audio/dai.
2. drop patch 1.

these host/dai code simply locate in src/audio looks not good, this is my start point.
let me know what do you think? @kv2019i @lgirdwood

File: src/ipc/ipc4/dai.c (hunk 1 above)
Comment (lgirdwood) [reply src/ipc/ipc4/dai.c:30]: This is good as its gives us more visibility on where we have stickyness around legacy abstractions.

**Review by lgirdwood — APPROVED** [review]
LGTM, pls remeber this is a next step (but not final) in seperation of module code from RTOS/IPC code.

File: src/ipc/ipc4/dai.c (hunk 1 above)
Comment (kv2019i) [reply src/ipc/ipc4/dai.c:30]: @btian1 @lgirdwood I'd still vote for "#include "../audio/copier" . and not "#include <.../audio/...". Mixing relative include paths and system includes is not clean -> https://gcc.gnu.org/onlinedocs/cpp/Include-Syntax.html

**Review by kv2019i — CHANGES_REQUESTED** [review]
Thanks @btian1 , this looks a lot cleaner. One small nit on using system-header syntax to refer to in-tree headers, see discussion inline.

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@kv2019i changes made.

**Review by kv2019i — APPROVED** [review]
Thanks @btian1 !

**Comment by kv2019i** [comment]
The Zephyr main fails are unrelated to this PR, proceeding with merge.

**Pull request merged**